docs/testing.md

```diff
@@ -38,8 +38,51 @@ declare_deps_provider(
 `junit_classpath_provider` (deps_id `junit_classpath`) is where classpath required for junit tests
 is defined.
 
-Toolchain must be registerd in your `WORKSPACE` file: 
+ScalaTest support can be enabled by configuring a provider with an id `scalatest_classpath`:
```

Comment:
unrelated to this change but this doc seems to suggest that in order to use scalatest I need to configure it. Why doesn't it show the easy defaults way?

Reply:
Not sure I understood your comment, there are no defaults when setting up custom toolchain

Reply:
Is `docs/testing.md` only about custom testing toolchains?

Reply:
Probably not, but I think it requires a lot more effort to rethink/rewrite whole setting up rules scala docs. I think it should be done outside this PR.

Reply:
My point- configuring custom deps became more complicated (but more powerful and we're ok with it).
If the readme before caused people to think they need to do X to use it now it seems like one needs to do 5X.
Can you please add a note here clarifying that this is needed only for the custom case?

Reply:
I have reorganized doc and added a note about custom deps

Reply:
Thanks. I appreciate it